Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs

```diff
@@ -87,65 +72,27 @@ protected override void AddChildView(VisualElement childView)
 				// FragmentContainer in order to get isolated fragment management
 				Fragment fragment = FragmentContainer.CreateInstance(page);
 				
-				var fc = fragment as FragmentContainer;
-
-				fc?.SetOnCreateCallback(pc =>
+				(fragment as FragmentContainer)?.SetOnCreateCallback(pc =>
 				{
 					_pageContainer = pc;
 					SetDefaultBackgroundColor(pc.Child);
 				});
 
-				FragmentTransaction transaction = FragmentManager.BeginTransaction();
-
-				if (_currentFragment != null)
-				{
-					transaction.Remove(_currentFragment);
-				}
-
-				transaction.Add(Id, fragment);
-				transaction.SetTransition((int)FragmentTransit.None);
-
-				// We don't currently support fragment restoration 
-				// So we don't need to worry about loss of this fragment's state
-				transaction.CommitAllowingStateLoss();
-
-				_currentFragment = fragment;
-
-				new Handler(Looper.MainLooper).PostAtFrontOfQueue(() =>
-				{
-					if (_pageContainer == null)
-					{
-						// The view we're hosting in the fragment was never created (possibly we're already 
-						// navigating to another page?) so there's nothing to commit
-						return;
-					}
-
-					FragmentManager.ExecutePendingTransactions();
-				});
+				SetCurrentFragment(fragment);
 			}
 		}
 
 		protected override void Dispose(bool disposing)
 		{
 			if (_disposed)
-			{
 				return;
-			}
 
 			_disposed = true;
 
 			if (disposing)
 			{
-				if (_currentFragment != null && !FragmentManager.IsDestroyed)
-				{
-					FragmentTransaction transaction = FragmentManager.BeginTransaction();
-					transaction.Remove(_currentFragment);
-					transaction.SetTransition((int)FragmentTransit.None);
-					transaction.CommitAllowingStateLoss();
-					FragmentManager.ExecutePendingTransactions();
```

Comment:
`ExecutePendingTransactions()` here was not being run inside a looper, but the refactored method is. Is this OK?